torch/csrc/autograd/input_buffer.h

```diff
@@ -37,7 +37,12 @@ struct InputBuffer {
   // Returns the inputs as a list of variables. Destroys given InputBuffer.
   static std::vector<Variable> variables(InputBuffer&& g);
 
+  c10::Stream getStreamForDeviceIdx(c10::Device device);
+
   std::vector<Variable> buffer;
+
+  inline static std::unordered_map<c10::DeviceIndex, c10::Stream>
+      device_streams;
```

Comment:
Should it be std::unordered_map<c10::Device, c10::Stream> in case there are different kinds of device on a single host?

Reply:
In the current state of accelerators, there can be only one attached to a process at any time. So it is ok.
If we ever change that I agree that this global state might be hard to find, so I'm also happy to index based on the Device here just in case as I don't expect it's going to be much more expensive.